decorator.rb

```diff
@@ -1,13 +1,13 @@
 require_relative 'Nameable'
 # Create decorator class called Decorator that inherits from Nameable.
 class Decorator < Nameable
-    def initialize(nameable)
-        @nameable = nameable
-        def correct_name
-            @nameable.correct_name
-        end
-    end
-    def correct_name
-        @nameable.correct_name
-    end
-end
\ No newline at end of file
+  super initialize(nameable)
+  @nameable = nameable
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+def correct_name
+  @nameable.correct_name
+end
```

Comment:
- [ ] Please add `attr_accessor: nameble`  and remember to close all methods well
```
class Decorator < Nameable
    attr_accessor :nameable
  
    def initialize(nameable)
      @nameable = nameable
    end
  

    def correct_name
      @nameable.correct_name
    end
end
    
```